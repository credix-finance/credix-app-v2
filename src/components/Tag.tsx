import React from "react";

type TagTypes = "active" | "pending" | "inactive" | "ended";
interface TagProps {
	type: TagTypes;
	children?: React.ReactNode;
}

const tagTypeStyles = {
	active: `bg-neutral-60 border-neutral-60 text-credix-primary`,
	inactive: `bg-neutral-10 border-neutral-10`,
	pending: `bg-neutral-40 border-neutral-40 text-credix-primary`,
	ended: `bg-credix-primary border-solid border-darker color-darker`,
};

export const Tag = ({ children, type }: TagProps) => {
	return (
		<div
			className={`py-[2px] border rounded-full min-w-[88px] font-semibold w-min flex justify-center ${tagTypeStyles[type]}`}
		>
			{children}
		</div>
	);
};
