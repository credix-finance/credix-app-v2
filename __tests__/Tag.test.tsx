import React from "react";
import renderer from "react-test-renderer";
import { Tag } from "@components/Tag";

test("Active tag", () => {
	const component = renderer.create(<Tag type="active">Active</Tag>);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Inactive tag", () => {
	const component = renderer.create(<Tag type="inactive">Inactive</Tag>);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Ended tag", () => {
	const component = renderer.create(<Tag type="ended">Ended</Tag>);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Pending tag", () => {
	const component = renderer.create(<Tag type="pending">Pending</Tag>);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
