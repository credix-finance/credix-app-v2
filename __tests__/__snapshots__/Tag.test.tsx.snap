// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active tag 1`] = `
<div
  className="py-[2px] px-3 border rounded-full min-w-[88px] font-semibold w-min flex justify-center bg-neutral-60 border-neutral-60 text-credix-primary"
>
  Active
</div>
`;

exports[`Ended tag 1`] = `
<div
  className="py-[2px] px-3 border rounded-full min-w-[88px] font-semibold w-min flex justify-center bg-credix-primary border-solid border-darker color-darker"
>
  Ended
</div>
`;

exports[`Inactive tag 1`] = `
<div
  className="py-[2px] px-3 border rounded-full min-w-[88px] font-semibold w-min flex justify-center bg-neutral-10 border-neutral-10"
>
  Inactive
</div>
`;

exports[`Pending tag 1`] = `
<div
  className="py-[2px] px-3 border rounded-full min-w-[88px] font-semibold w-min flex justify-center bg-neutral-40 border-neutral-40 text-credix-primary"
>
  Pending
</div>
`;
